lib/encoder/src/AEAT6012.cpp

```diff
@@ -95,18 +95,19 @@ bool AEAT6012::read(void) {
   }
 }
 
-bool AEAT6012::getAngleDeg(float &angle) {
+bool AEAT6012::update() {
```

Comment:
@cindyli-13 did i break ur driver anywhere

Reply:
Changes look fine, only recommendation is to just rename the read() function to update(). Also I'm not sure if we want to keep the getAngleDegNoTrigger() and getAngularVelocityDegPerSecNoTrigger() since getAngleDeg() and getAngularVelocityDegPerSec() now do the same thing. The only concern is the mutex locking - we can't use them in ISR contexts if we eventually want to do that with the async driver.

Reply:
ive removed the notrigger functions. ive renamed your readAsync to update since the fact that it takes in a callback heavily implies that. i cant rename read to update bc there will be a name conflict since itd have the same signature as an existing function

Reply:
I meant instead of having `update()` be:
```
bool AEAT6012::update() {
  std::scoped_lock<Mutex> lock(m_mutex);
  return read();
}
```
just put the body of `read()` in this `update()` function. I don't think there's a name conflict since there's already 2 overloaded `update()` functions, one takes in no parameters and the other takes in a callback. The only problem might be that `read()` is called in `reset()`, idk if the scoped mutex is recursive. If the recursive mutex locking is a problem then it's fine the way it is rn